Mathlib/SetTheory/Ordinal/Nimber.lean

```diff
@@ -356,4 +366,33 @@ theorem add_trichotomy {a b c : Nimber} (h : a + b + c ≠ 0) :
   · rw [← hx'] at hx
     exact Or.inr <| Or.inr hx
 
+/-- Nimber addition of naturals corresponds to the bitwise XOR operation. -/
+theorem add_nat (a b : ℕ) : ∗a + ∗b = ∗(a ^^^ b) := by
+  apply le_antisymm
+  · apply add_le_of_forall_ne
+    all_goals
+      refine Nimber.rec (fun c hc ↦ ?_)
+      rw [OrderIso.lt_iff_lt] at hc
+      obtain ⟨c, rfl⟩ := Ordinal.lt_omega0.1 (hc.trans (Ordinal.nat_lt_omega0 _))
```

Comment:
You use this pattern twice - maybe it should be a lemma? If an ordinal is <= a natural, then it's equal to a natural.

Reply:
I've added `eq_nat_of_le_nat`, it does simplify the proof a bit.